src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -2319,8 +2319,7 @@ public static void Refresh(Assembly assembly)
         [EditorBrowsable(EditorBrowsableState.Advanced)]
         public static Type ComObjectType
         {
-            [PreserveDependency(".ctor", "System.ComponentModel.TypeDescriptor/TypeDescriptorComObject")]
-            [PreserveDependency(".ctor", "System.ComponentModel.TypeDescriptor/ComNativeDescriptorProxy")] // TODO: https://github.com/mono/linker/issues/801
+            [DynamicallyAccessedMembersAttribute(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
```

Comment:
@MichalStrehovsky - should this `DynamicallyAccessedMembersAttribute` go on the property itself or just the `getter`? In your comment you mentioned it should be `[return: DynamicallyAccessedMembersAttribute`, but we don't have the `return:` here.

Reply:
It can either go on the property, or on the return parameter of the getter. Placing it on the getter method is a bug.

It's exactly the kind of bug @vitek-karas really wanted to avoid, but was forced to introduce in #36340 when we allowed DynamicallyAccessedMembers on methods only because we need it to annotate Type/RuntimeType/TypeInfo.